app/services/jwt/decode_service.rb

```diff
@@ -0,0 +1,109 @@
+module JWT
+  class DecodeService
```

Comment:
The structure of this is a little confusing. Do these classes need to be split out?

Reply:
I did started it out as `EncodeService` & `DecodeService`...

> EncodeService belongs nested in DecodeService that is by designed as it has very little to no reason to live.

`EncodeService` was mainly introduced to support testing, and to support legacy `mcb` token generation, it more or less the mirror and the younger sibling of `DecodeService`.

ie in order to decode something needs to be encoded first,

Reply:
Discussed offline.

Reply:
I don't understand what's going on here either.

Reply:
Moved to MCB side